state-chain/runtime/src/chainflip/simulate_swap.rs

```diff
@@ -0,0 +1,232 @@
+			match additional_order {
+				SimulateSwapAdditionalOrder::LimitOrder {
+					base_asset,
+					quote_asset,
+					side,
+					tick,
+					sell_amount,
+				} => {
+					LiquidityPools::try_add_limit_order(
+						&AccountId::new([0; 32]),
+						base_asset,
+						quote_asset,
+						side,
+						index as OrderId,
+						tick,
+						sell_amount.into(),
+					)?;
+				},
+			}
+		}
+	}
+
+	fn remove_fees(
+		ingress_or_egress: IngressOrEgress,
+		asset: Asset,
+		amount: AssetAmount,
+	) -> (AssetAmount, AssetAmount) {
+		use pallet_cf_ingress_egress::AmountAndFeesWithheld;
+
+		match asset.into() {
+			ForeignChainAndAsset::Ethereum(asset) => {
+				let AmountAndFeesWithheld {
+							amount_after_fees,
+							fees_withheld,
+						} = pallet_cf_ingress_egress::Pallet::<Runtime, EthereumInstance>::withhold_ingress_or_egress_fee(ingress_or_egress, asset, amount.unique_saturated_into());
+
+				(amount_after_fees, fees_withheld)
+			},
+			ForeignChainAndAsset::Polkadot(asset) => {
+				let AmountAndFeesWithheld {
+							amount_after_fees,
+							fees_withheld,
+						} = pallet_cf_ingress_egress::Pallet::<Runtime, PolkadotInstance>::withhold_ingress_or_egress_fee(ingress_or_egress, asset, amount.unique_saturated_into());
+
+				(amount_after_fees, fees_withheld)
+			},
+			ForeignChainAndAsset::Bitcoin(asset) => {
+				let AmountAndFeesWithheld {
+							amount_after_fees,
+							fees_withheld,
+						} = pallet_cf_ingress_egress::Pallet::<Runtime, BitcoinInstance>::withhold_ingress_or_egress_fee(ingress_or_egress, asset, amount.unique_saturated_into());
+
+				(amount_after_fees.into(), fees_withheld.into())
+			},
+			ForeignChainAndAsset::Arbitrum(asset) => {
+				let AmountAndFeesWithheld {
+							amount_after_fees,
+							fees_withheld,
+						} = pallet_cf_ingress_egress::Pallet::<Runtime, ArbitrumInstance>::withhold_ingress_or_egress_fee(ingress_or_egress, asset, amount.unique_saturated_into());
+
+				(amount_after_fees, fees_withheld)
+			},
+			ForeignChainAndAsset::Solana(asset) => {
+				let AmountAndFeesWithheld {
+							amount_after_fees,
+							fees_withheld,
+						} = pallet_cf_ingress_egress::Pallet::<Runtime, SolanaInstance>::withhold_ingress_or_egress_fee(ingress_or_egress, asset, amount.unique_saturated_into());
+
+				(amount_after_fees.into(), fees_withheld.into())
+			},
+			ForeignChainAndAsset::Assethub(asset) => {
+				let AmountAndFeesWithheld {
+							amount_after_fees,
+							fees_withheld,
+						} = pallet_cf_ingress_egress::Pallet::<Runtime, AssethubInstance>::withhold_ingress_or_egress_fee(ingress_or_egress, asset, amount.unique_saturated_into());
+
+				(amount_after_fees, fees_withheld)
+			},
+		}
+	}
+
+	let include_fee = |fee_type: FeeTypes| !exclude_fees.contains(&fee_type);
+
+	// Default to using the DepositChannel fee unless specified.
+	let (amount_to_swap, ingress_fee) = if include_fee(FeeTypes::IngressDepositChannel) {
```

Comment:
Maybe subjective, but I find the `include_fee`/`remove_fees`/`exclude_fees` terminology very confusing. I'm assuming `include_fee` means the fee will be appied and `remove_fee` is you applying the fee? How about `charge_fee` instead of `remove_fee` (otherwise `remove` and `include` sounds like they have opposite meaning)?